Mob 407 Event details enhancement - Attendee List

YagoAccenture opened the pull request.

YagoAccenture committed:
Event details screen completed with the logic to show or hide the button

YagoAccenture committed:
MOB-407 The assistant screen displays the data.

YagoAccenture committed:
Merge branch 'main' into MOB-407-Flutter

YagoAccenture committed:
Merge branch 'main' into MOB-407-Flutter

YagoAccenture committed:
MOB-407 VerifyAccountCubit duplication removed after main merge

YagoAccenture committed:
MOB-407 Created tests for attendees cubit and added 2 tests to details cubit

The navigation logic was also moved from the screen to the cubit

PabloGrelaM commented on lib/datasource_remote/event/event_attendees_remote_model.dart:
```diff
@@ -0,0 +1,11 @@
+
+  factory EventAttendeeRemoteModel.fromJson(Map<String, dynamic> json) {
+    return EventAttendeeRemoteModel(
+      name: json['name'] ?? '',
```
name is mandatory

PabloGrelaM commented on lib/domain/event/model/event_attendees.dart:
```diff
@@ -0,0 +1,17 @@
+
+  EventAttendees({required this.totalUsers, required this.users});
+
+  factory EventAttendees.fromJson(Map<String, dynamic> json) {
```
this mapper shuldn't be here, this is already in the event attendee remote model

PabloGrelaM commented on lib/view/events/event_attendees/mapper/event_attendees_ui_mapper.dart:
```diff
@@ -0,0 +1,20 @@
+import 'package:esmorga_flutter/domain/event/model/event_attendees.dart';
+
+class EventAttendeesUiModel {
+  final int totalUsers;
```
this totalUsers is not used by the screen as you are checking here the hasAttendees, you don't need to send it.

PabloGrelaM commented on lib/view/events/event_attendees/view/event_attendees_screen.dart:
```diff
@@ -0,0 +1,91 @@
+                    final userName = attendees.users[index];
+                    return Padding(
+                      padding: const EdgeInsets.only(bottom: 17.0),
+                      child: EsmorgaText(
```
use the EsmorgaRow component.

PabloGrelaM commented on lib/view/events/event_detail/view/event_detail_screen.dart:
```diff
@@ -45,6 +45,13 @@ class _EventDetailFormState extends State<_EventDetailForm> {
+      } else if (effect is NavigateToLoginEffect) {
+        widget.goToLogin();
+      } else if (effect is NavigateToAttendeesEffect) {
+        context.push('/event_attendees/${effect.eventId}');
```
the navigation should be in the app_routes and then passed through a lambda param to the screen.

PabloGrelaM commented on lib/view/events/event_detail/view/event_detail_screen.dart:
```diff
@@ -158,25 +165,32 @@ class _EventDetailFormState extends State<_EventDetailForm> {
-                  text: l10n.labelCapacity(
-                    ui.currentAttendeeCount,
-                    ui.maxCapacity!,
+                if (ui.maxCapacity != null && ui.maxCapacity! > 0) ...[
```
you are checking twice this, I don't think you need the previous if.

YagoAccenture committed:
MOB 407 PR comments solved

The name is now required in the event_attendees_remote_model.

The mapper is now only in the event_attendees_remote_model.

TotalUsers has been removed from the ui_mapper.

The navigation has been corrected.

Duplicate logic has been removed from the event_details_screen.

And an esmorga_checkbox is used to display the attendees' names.

YagoAccenture committed:
MOB 407 test corrected totalUsers is no needed now

PabloGrelaM commented on lib/data/event/model/event_attendees_data_model.dart:
```diff
@@ -0,0 +1,18 @@
+    required this.users,
+  });
+
+  factory EventAttendeesDataModel.fromJson(Map<String, dynamic> json) {
```
where are you using this fromJson model?

YagoAccenture committed:
MOB-407 Unused fromjson deleted

YagoAccenture committed:
MOB-407 Remove fromJson mapper from EventAttendees

YagoAccenture committed:
MOB 407 Deleted duplicated logic

YagoAccenture committed:
Corrected logic for displaying the number of attendees
